core/kazoo_json/src/kz_json.erl

```diff
@@ -1183,15 +1183,15 @@ is_private_key(<<"_", _/binary>>) -> 'true';
 is_private_key(<<"pvt_", _/binary>>) -> 'true';
 is_private_key(_) -> 'false'.
 
--type composite_key() :: binary() | [binary()].
+-type composite_key() :: key() | [keys()].
```

Comment:
No, path() includes array indicies (integers). Flatten doesn't go deeper if its not a json object.